rotkehlchen/db/accounting_rules.py

```diff
@@ -0,0 +1,140 @@
+from typing import TYPE_CHECKING, Optional
+
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.evm.accounting.structures import BaseEventSettings
+from rotkehlchen.db.constants import NULL_ACCOUNTING_COUNTERPARTY
+from rotkehlchen.db.filtering import AccountingRulesFilterQuery
+from rotkehlchen.errors.misc import InputError
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+class DBAccountingRules:
+
+    def __init__(self, db_handler: 'DBHandler') -> None:
+        self.db = db_handler
+
+    @classmethod
+    def _formulate_primary_key(
+            cls,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            counterparty: Optional[str],
+    ) -> tuple[str, str, str]:
+        return (
+            event_type.serialize(),
+            event_subtype.serialize(),
+            counterparty if counterparty is not None else NULL_ACCOUNTING_COUNTERPARTY,
+        )
+
+    def add_accounting_rule(
+            self,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            counterparty: Optional[str],
+            rule: 'BaseEventSettings',
+    ) -> None:
+        query = (
+            'INSERT INTO accounting_rules(type, subtype, counterparty, taxable, '
+            'count_entire_amount_spend, count_cost_basis_pnl, method, accounting_treatment) '
+            'VALUES (?, ?, ?, ?, ?, ?, ?, ?)'
+        )
+        with self.db.conn.write_ctx() as write_cursor:
+            try:
+                write_cursor.execute(
+                    query,
```

Comment:
same as before. Don't use a `query` variable if it's only used in the query itself once
```suggestion
        with self.db.conn.write_ctx() as write_cursor:
            try:
                write_cursor.execute(
                    'INSERT INTO accounting_rules(type, subtype, counterparty, taxable, '
                    'count_entire_amount_spend, count_cost_basis_pnl, method, accounting_treatment) '
                    'VALUES (?, ?, ?, ?, ?, ?, ?, ?)'
```